docs/nic-health-monitor/syslog-detection-correlation.md

```
@@ -477,6 +483,57 @@ journalctl -k | grep -E "(mlx5_core|PCIe)"
+regex can match both, so this pattern is implemented as a small stateful
+detector inside the NIC driver handler rather than as an entry in the
+first-match-wins single-line loop:
+
+1. **Arm**: a line matching `BUG: soft lockup - CPU#(\d+) stuck for (\d+)s`
+   opens a bounded window (150 kernel log lines). A newer header re-arms the
+   window and refreshes the captured CPU/duration (the kernel re-reports a
+   persistent lockup with a growing duration).
+2. **Confirm**: within the window, a line matching
+   `mlx5e_(poll_ico_cq|napi_poll)\+0x` — the `RIP:` line or a call-trace
+   frame — confirms mlx5 attribution and emits one **Fatal** event
+   (`REPLACE_VM`). Frames prefixed with `? ` are speculative stack remnants
```

markdownlint-cli2 (0.23.0)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)